app/backend/src/couchers/servicers/notifications.py

```diff
@@ -102,25 +103,25 @@ def GetVapidPublicKey(self, request, context):
 
         return notifications_pb2.GetVapidPublicKeyRes(vapid_public_key=get_vapid_public_key())
 
-    def RegisterPushNotification(self, request, context):
+    def RegisterPushNotificationSubscription(self, request, context):
         if not config["PUSH_NOTIFICATIONS_ENABLED"]:
             context.abort(grpc.StatusCode.UNAVAILABLE, errors.PUSH_NOTIFICATIONS_DISABLED)
 
         with session_scope() as session:
+            data = json.loads(request.full_subscription_json)
             subscription = PushNotificationSubscription(
                 user_id=context.user_id,
-                endpoint=request.endpoint,
-                p256dh_key=decode_key(request.p256dh_key),
-                auth_key=decode_key(request.auth_key),
+                endpoint=data["endpoint"],
+                p256dh_key=decode_key(data["keys"]["p256dh"]),
+                auth_key=decode_key(data["keys"]["auth"]),
                 full_subscription_info=request.full_subscription_json,
                 user_agent=request.user_agent,
             )
             session.add(subscription)
             session.flush()
-            sub_id = subscription.id
             push_to_subscription(
```

Comment:
how this works exactly? it's an http sent to the frontend and the frontend it's listening? (through the service worker?)

or is this some websocket connection? (just to understand what's going on)